src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -86,21 +86,24 @@ public void discoverClusterSlots(Jedis jedis) {
       rediscovering = true;
 
       try {
-        this.slots.clear();
-
         List<Object> slots = jedis.clusterSlots();
 
+        // We should clear slots after getting result about cluster slots, because if we got exception or timeout in
```

Comment:
> JedisSlotBasedConnectionHandler#getConnectionFromSlot, this comment becomes invalid.
Not completely.
1) First thread started to discover. Clear nodes. Release lock while going to next node.
2) On this time second thread went to JedisSlotBasedConnectionHandler#getConnectionFromSlot, very possible that we could just use old pool, but we see empty collection and connect to discovering party.
I would prefer to don't publish empty collection while we can leave old one.

> And IMHO, clearing slots first or later is up to amount of slot changeset
I'm not sure if I get you. We clear it completely anyways. I just moved clearing after external code. So, to avoid situation of exception in external code, which leads to publishing empty slots collection.